plugins/portal-history-ornaments.js

```diff
@@ -0,0 +1,244 @@
+  portalsHistory.iconMarked = {};
+	portalsHistory.iconScouted = {};
+  var parts = (portalsHistory.settings.showVisitedCaptured + portalsHistory.settings.showScouted);
+  LEVEL_TO_RADIUS.forEach((el, idx) => {
+    var size = (el * 2 + 8) * scale;
+    var offset = 0;
+    if (portalsHistory.settings.showScouted) {
+      portalsHistory.iconScouted[idx] = svgToIcon(getSVGString(size, 'violet', parts, offset), size + 4);
+    offset++;
+      } else {
+      portalsHistory.iconScouted[idx] = svgToIcon(getSVGString(size, 'transparent', parts, offset), size + 4);
+    }
+
+    if (portalsHistory.settings.showVisitedCaptured) {
+      portalsHistory.iconSemiMarked[idx] = svgToIcon(getSVGString(size, 'yellow', parts, offset), size + 4);
+      portalsHistory.iconMarked[idx] = svgToIcon(getSVGString(size, 'red', parts, offset), size + 4);
+      offset++;
+    } else {
+      portalsHistory.iconSemiMarked[idx] = svgToIcon(getSVGString(size, 'transparent', parts, offset), size + 4);
+      portalsHistory.iconMarked[idx] = svgToIcon(getSVGString(size, 'transparent', parts, offset), size + 4);
+    }
+  });
+}
+
+function drawPortalFlags (portal) {
+  var drawMissing = portalsHistory.settings.drawMissing;
+  portal._historyLayer = new L.LayerGroup();
+
+  if (portal.options.data.history) {
+    if (drawMissing && !portal.options.data.history.visited || !drawMissing && portal.options.data.history.captured) {
+      L.marker(portal._latlng, {
+        icon: portalsHistory.iconMarked[portal.options.level],
+        interactive: false,
+        keyboard: false,
+      }).addTo(portal._historyLayer);
+    }
+    if (drawMissing && portal.options.data.history.visited && !portal.options.data.history.captured 
+        || !drawMissing && portal.options.data.history.visited && !portal.options.data.history.captured) {
+      L.marker(portal._latlng, {
+        icon: portalsHistory.iconSemiMarked[portal.options.level],
+        interactive: false,
+        keyboard: false,
+      }).addTo(portal._historyLayer);
+    }
+    if (drawMissing && !portal.options.data.history.scoutControlled || !drawMissing && portal.options.data.history.scoutControlled) {
+      L.marker(portal._latlng, {
+        icon: portalsHistory.iconScouted[portal.options.level],
+        interactive: false,
+        keyboard: false,
+      }).addTo(portal._historyLayer);
+    }
+  }
+  portal._historyLayer.addTo(portalsHistory.layerGroup);
+}
+
+function drawAllFlags () {
+  portalsHistory.layerGroup.clearLayers();
+  createIcons();
+  //IITC.me support: getCurrentZoomTileParameters is iitc.app only; iitc.me function is: getMapZoomTileParameters
+  var tileParams = window.getCurrentZoomTileParameters ? window.getCurrentZoomTileParameters() : window.getMapZoomTileParameters();
+  if (tileParams.level !== 0) {
+    return;
+  }
+
+  for (var id in window.portals) {
+    drawPortalFlags(window.portals[id]);
+  }
+}
+
+function getSVGString (size, color, parts, offset) {
```

Comment:
Such SVG usage is not optimal. I will return to it later.

Reply:
revamped